src/hooks/usePost.ts

```diff
@@ -39,7 +48,7 @@ export const usePost = <
                     });
             });
         },
-        [apiActionHandler, dispatch],
+        [apiActionHandler, dispatch, headers],
```

Comment:
@JuraJuki Check the annotation of the PR below this comment. It says that it has a missing dependency!